File: backend/internal/psql/audit.go
package psql

import "time"

type Timestamps struct {
	CreatedAt time.Time `db:"created_at"`
	UpdatedAt time.Time `db:"updated_at"`
}

type AuditFields struct {
	CreatedBy int64 `db:"created_by"`
	UpdatedBy int64 `db:"updated_by"`
}

File: frontend/src/lib/audit.ts
import { z } from "zod";
import { idSchema } from "./api";

export const auditDates = z.object({
  createdAt: z.coerce.date(),
  updatedAt: z.coerce.date(),
});

export const auditMembers = z.object({
  createdBy: idSchema,
  updatedBy: idSchema,
});
